sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -343,18 +343,17 @@ function setName(entry: any, nameProperties: any): any {
  * @param response
  */
 export function buildError(errorBody: any, response: HttpOperationResponse): RestError {
-  const errorProperties = errorBody.Error || errorBody.error || errorBody;
   let errorMessage;
 
   if (typeof errorBody === "string") {
     errorMessage = errorBody;
   } else {
-    errorMessage = errorProperties.Detail || errorProperties.detail;
+    errorMessage = errorBody.Error.Detail;
   }
 
   const error: RestError = new RestError(
     errorMessage,
-    errorProperties.Code || errorProperties.code,
+    "SERVICE_ERROR",
```

Comment:
I will defer to @bterlson on this.

Reply:
@bterlson: Goal is for users to be able to differentiate errors as well as they can in .NET SDK and not necessarily replicate if can be avoided

Replicate the if-else checks on Code/statusCode that .NET SDK uses - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/ManagementClient.cs#L915

Can add unit tests and avoid live tests

Reply:
The other concern we discussed was about how `MessagingError` in Service Bus relies on `name` property to differentiate errors instead. Filed issue https://github.com/Azure/azure-sdk-for-js/issues/6611 for it. For current PR, we can go with as decided on setting the code to a value that's useful for end users reference.